lib/flight_scheduler/commands/info.rb

```diff
@@ -28,43 +28,291 @@
+        end
+
+        def register_default_columns
+          register_partition_name
+          register_column(header: 'AVAIL') { |_| 'TBD' }
+          register_column(header: 'TIMELIMIT') { |_| 'TBD' }
+          register_node_count
+          register_node_state
+          register_nodelist
+        end
+
+        private
+
+        # ///////////////////////////////////////////////////////////
+        # Main helper methods for selecting column type:
+        # Each column can render: a partition, a node, or many nodes
+        # A table can not contain both a partition and a node column
+        # however the "nodes" can be shared
+        def register_partition_column(**opts, &b)
+          @partition_basis = true
+          register_column(**opts) do |partition, _n|
+            b.call(partition)
+          end
+        end
+
+        def register_node_column(**opts, &b)
+          @node_basis = true
+          register_column(**opts) do |partition, nodes|
+            b.call(nodes.first)
+          end
+        end
+
+        def register_nodes_column(**opts, &b)
+          register_column(**opts) do |partition, nodes|
+            nodes ||= (partition.nodes || [])
+            b.call(nodes)
+          end
+        end
+        # End main helpers
+        # The following register an actual column
+        # ///////////////////////////////////////////////////////////
+
+        def register_partition_name
+          register_partition_column(header: 'PARTITION') { |p| p.name }
+        end
+
+        def register_node_count
+          register_nodes_column(header: 'NODES') { |n| n.length }
+        end
+
+        # Assumes all the nodes have been grouped by state
+        def register_node_state
+          @node_state = true
+          register_nodes_column(header: 'STATE') do |nodes|
+            nodes.first&.state
+          end
+        end
+
+        def register_nodelist
+          register_nodes_column(header: 'NODELIST') { |n| n.map(&:name).join(',') }
+        end
+
+        # NOTE: This method is used to quasi-list the nodes instead of partitions
+        #
+        # It assumes a one-to-one mapping of partitions to nodes. This is achieved using
+        # the PartitionProxy and does not represent the underlining data model
+        def register_hostnames
+          register_node_column(header: 'HOSTNAMES') { |n| n.name }
+        end
+
+        def register_cpus
+          register_nodes_column(header: 'CPUS') do |nodes|
+            value_or_min_plus(*nodes.map(&:cpus))
+          end
+        end
+
+        def register_gpus
+          register_nodes_column(header: 'GPUS') do |nodes|
+            value_or_min_plus(*nodes.map(&:gpus))
+          end
+        end
+
+        def register_memory
+          register_nodes_column(header: 'MEMORY (MB)') do |nodes|
+            value_or_min_plus(*nodes.map(&:gpus)) do |value|
+              # Convert the memory into MB
+              sprintf('%.2f', value.fdiv(1048576))
```

Comment:
This should be `&:memory` not `&:gpus`.

Also, lets format it as `%d`.  That way running `free -m` and `scheduler info` both agree.

Reply:
https://github.com/openflighthpc/flight-scheduler/pull/19/commits/1fc828da8494403b9cb36f8d76f4a835783bf3df